### PR TITLE
[tune] Support TF2.0 on Keras Callback

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Make keras callback 2.0 compatible

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: python/ray/tune/integration/keras.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 from __future__ import division
 from __future__ import print_function
 
-import keras
+import tensorflow.keras as keras
```
Comment (hartikainen) [inline python/ray/tune/integration/keras.py]:
```suggestion
from tensorflow import tensorflow
```

File: python/ray/tune/integration/keras.py [hunk 2]
```diff
@@ -45,4 +44,7 @@ def on_epoch_end(self, batch, logs={}):
         for metric in list(logs):
             if "loss" in metric and "neg_" not in metric:
                 logs["neg_" + metric] = -logs[metric]
-        self.reporter(keras_info=logs, mean_accuracy=logs["acc"])
+        if "acc" in logs:
```
Comment (hartikainen) [inline python/ray/tune/integration/keras.py]: How does this thing depend on tf version?

File: python/ray/tune/integration/keras.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/tune/integration/keras.py]: from tensorflow import tensorflow :) 

anyways, fixed to keras

File: python/ray/tune/integration/keras.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/tune/integration/keras.py]: for some reason keras changes their result dictionary when moving to tensorflow.keras.

**Review by hartikainen — APPROVED** [review]

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by richardliaw** [comment]
jenkins test tune

**Comment by prob1995** [comment]
@richardliaw  @hartikainen  
Maybe you can also consider to update ray/tune/examples/utils.py to fit tf2.0 without installing keras, because utils.py will be used in examples/tune_mnist_keras.py 

For example, in utils.py line 6~8 :
import tensorflow.keras as keras
from tensorflow.keras.datasets import mnist
from tensorflow.keras import backend as K

**Comment by richardliaw** [comment]
@prob1995 I'm adding a TF2 Keras example separately - #5898

**Pull request merged**